### ISSUE TITLE
manifest for <container> not found problem persists across restart/reboot

### ISSUE THREAD

**Issue by radoye** [Azure/iotedge#1378]
## Expected Behavior

Occasionally we get the following message in the logs

``` bash
5804 Jun 23 22:27:12 hopper iotedged[8282]: 2019-06-24T05:27:12Z [INFO] - Pulling image iotdevreg.azurecr.io/beluga/heartrate.cpu:latest...
5805 Jun 23 22:27:18 hopper iotedged[8282]: 2019-06-24T05:27:18Z [WARN] - Could not pull image iotdevreg.azurecr.io/beluga/heartrate.cpu:latest                                                                
5806 Jun 23 22:27:18 hopper iotedged[8282]: 2019-06-24T05:27:18Z [WARN] -         caused by: manifest for iotdevreg.azurecr.io/beluga/heartrate.cpu:latest not found 
```

It seems to correlate with connectivity issues on one of our test machines.

I would expect the problem to resolve after the connectivity is restored, or after reboot or iotedge restart. It persists.

## Current Behavior

The problem persists across reboot & iotedge restart. The only way to deal with this is to remove the module in deployment, then reconfigure the same exact module deployment and update. Only after that the expected operation resumes.

## Steps to Reproduce

Unsure. 

### Output of `iotedge check`

This was done after the problem was resolved. 
**NOTE**: we have to use Docker as our modules need access to NVidia GPUs.

<details>
<summary>Click here</summary>

```
Configuration checks
--------------------
√ config.yaml is well-formed
√ config.yaml has well-formed connection string
√ container engine is installed and functional
√ config.yaml has correct hostname
√ config.yaml has correct URIs for daemon mgmt endpoint
√ latest security daemon
√ host time is close to real time
√ container time is close to host time
‼ DNS server
    Container engine is not configured with DNS server setting, which may impact connectivity to IoT Hub.
    Please see https://aka.ms/iotedge-prod-checklist-dns for best practices.
    You can ignore this warning if you are setting DNS server per module in the Edge deployment.
‼ production readiness: certificates
    Device is using self-signed, automatically generated certs.
    Please see https://aka.ms/iotedge-prod-checklist-certs for best practices.
√ production readiness: certificates expiry
‼ production readiness: container engine
    Device is not using a production-supported container engine (moby-engine).
    Please see https://aka.ms/iotedge-prod-checklist-moby for details.
‼ production readiness: logs policy
    Container engine is not configured to rotate module logs which may cause it run out of disk space.
    Please see https://aka.ms/iotedge-prod-checklist-logs for best practices.
    You can ignore this warning if you are setting log policy per module in the Edge deployment.

Connectivity checks
-------------------
√ host can connect to and perform TLS handshake with IoT Hub AMQP port
√ host can connect to and perform TLS handshake with IoT Hub HTTPS port
√ host can connect to and perform TLS handshake with IoT Hub MQTT port
√ container on the default network can connect to IoT Hub AMQP port
√ container on the default network can connect to IoT Hub HTTPS port
√ container on the default network can connect to IoT Hub MQTT port
√ container on the IoT Edge module network can connect to IoT Hub AMQP port
√ container on the IoT Edge module network can connect to IoT Hub HTTPS port
√ container on the IoT Edge module network can connect to IoT Hub MQTT port
√ Edge Hub can bind to ports on host

One or more checks raised warnings. Re-run with --verbose for more details.

```
</details>

### Device (Host) Operating System

Ubuntu 18.04

### Architecture

amd64

### Container Operating System

Linux containers.

#### iotedged

iotedge 1.0.7.1 (f7c51d92be8336bc6be042e1f1f2505ba01679f3)

#### Edge Agent

mcr.microsoft.com/azureiotedge-agent:1.0

#### Edge Hub

mcr.microsoft.com/azureiotedge-hub:1.0

#### Docker

```
Client:
 Version:           18.09.6
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        481bc77
 Built:             Sat May  4 02:35:57 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.6
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       481bc77
  Built:            Sat May  4 01:59:36 2019
  OS/Arch:          linux/amd64
  Experimental:     false

```

[iotedge logs pastebin](https://paste.ubuntu.com/p/f6yGGFmcW3/)

[edge-agent logs pastebin](https://paste.ubuntu.com/p/298N8bhKCK/)

[edge-hub logs pastebin](https://paste.ubuntu.com/p/xr7ZRXnCmc/)

**Comment by darobs** [comment]
Hello @radoye 

Interesting problem.  I'm surprised the problem persists over a reboot. I will have to do some digging before I know what's going on with this.  I will update this issue when I find something out.

Thanks,
Dave

**Comment by arsing** [comment]
The error comes from Docker. `iotedge restart` won't do anything for that.

If it persists even after restoring connectivity, then try `docker pull iotdevreg.azurecr.io/beluga/heartrate.cpu:latest`

It might not be related to loss of network connectivity at all. It might be running out of disk space and a reboot happens to clean enough disk space to let it work.

**Comment by radoye** [comment]
@arsing Unless there is some configuration option to limit the space, I doubt the space can be an issue. The machine on which this happens has ~300G of free space.

**Comment by lt72** [comment]
@radoye apologies for the late answer. Are you still experiencign this issues?

**Comment by radoye** [comment]
@lt72 not lately. haven't really tried to repro, though.

**Comment by lt72** [comment]
@radoye I'll close as resolved, but please re-open as needed.
